test/pardot-test-srv/main.go

```diff
@@ -0,0 +1,256 @@
+	// that the token has been manually expired.
+	active string
+}
+
+type createdContacts map[string]struct{}
+
+type contacts struct {
+	sync.Mutex
+
+	created createdContacts
+}
+
+type testServer struct {
+	expectedClientID     string
+	expectedClientSecret string
+	token                token
+	contacts             contacts
+	developmentMode      bool
+}
+
+// generateToken generates a new random token.
+func generateToken() string {
+	bytes := make([]byte, 32)
+	for i := range bytes {
+		bytes[i] = byte(rand.IntN(256))
+	}
+	return fmt.Sprintf("%x", bytes)
+}
+
+func (ts *testServer) getTokenHandler(w http.ResponseWriter, r *http.Request) {
+	err := r.ParseForm()
+	if err != nil {
+		http.Error(w, "Invalid request", http.StatusBadRequest)
+		return
+	}
+
+	clientID := r.FormValue("client_id")
+	clientSecret := r.FormValue("client_secret")
+
+	if clientID != ts.expectedClientID || clientSecret != ts.expectedClientSecret {
+		http.Error(w, "Invalid credentials", http.StatusUnauthorized)
+		return
+	}
+
+	ts.token.Lock()
+	defer ts.token.Unlock()
+	if ts.token.active == "" {
+		ts.token.active = generateToken()
+	}
+
+	response := map[string]interface{}{
+		"access_token": ts.token.active,
+		"token_type":   "Bearer",
+		"expires_in":   3600,
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	err = json.NewEncoder(w).Encode(response)
+	if err != nil {
+		log.Printf("Failed to encode token response: %v", err)
+		http.Error(w, "Failed to encode token response", http.StatusInternalServerError)
+	}
+}
+
+func (ts *testServer) createContactsHandler(w http.ResponseWriter, r *http.Request) {
+	ts.token.Lock()
+	validToken := ts.token.active
+	ts.token.Unlock()
+
+	token := r.Header.Get("Authorization")
+	businessUnitId := r.Header.Get("Pardot-Business-Unit-Id")
+
+	if token != "Bearer "+validToken {
+		http.Error(w, "Unauthorized", http.StatusUnauthorized)
+		return
+	}
+
+	if businessUnitId == "" {
+		http.Error(w, "Missing 'Pardot-Business-Unit-Id' header", http.StatusBadRequest)
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Failed to read request body", http.StatusInternalServerError)
+		return
+	}
+
+	type contactData struct {
+		Email string `json:"email"`
+	}
+
+	var contact contactData
+	err = json.Unmarshal(body, &contact)
+	if err != nil {
+		http.Error(w, "Failed to parse request body", http.StatusBadRequest)
+		return
+	}
+
+	if contact.Email == "" {
+		http.Error(w, "Missing 'email' field in request body", http.StatusBadRequest)
+		return
+	}
+
+	if ts.developmentMode {
+		ts.contacts.Lock()
+		_, exists := ts.contacts.created[contact.Email]
+		if !exists {
+			ts.contacts.created[contact.Email] = struct{}{}
+		}
+		ts.contacts.Unlock()
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	err = json.NewEncoder(w).Encode(map[string]string{"status": "success"})
+	if err != nil {
+		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+		return
+	}
```

Comment:
Creating a JSON encoder seems slightly overkill for a fixed string.

```
w.Write([]byte(`{"status": "success"}`))
```

Reply:
Good call.